network/p2p/inspector/internal/cache/cache_test.go

```diff
@@ -0,0 +1,549 @@
+	}
+}
+
+// TestRecordCache_ConcurrentSameRecordInit tests the concurrent initialization of the same record.
+// The test covers the following scenarios:
+// 1. Multiple goroutines attempting to initialize the same record concurrently.
+// 2. Only one goroutine successfully initializes the record, and others receive false on initialization.
+// 3. The record is correctly initialized in the cache and can be retrieved using the Get method.
+func TestRecordCache_ConcurrentSameRecordInit(t *testing.T) {
+	cache := cacheFixture(t, 100, defaultDecay, zerolog.Nop(), metrics.NewNoopCollector())
+	require.NotNil(t, cache)
+	// expect cache to be initialized with a empty active cluster IDs list
+	require.Equalf(t, uint(0), cache.Size(), "cache size must be 1")
+
+	nodeID := unittest.IdentifierFixture()
+	const concurrentAttempts = 10
+
+	var wg sync.WaitGroup
+	wg.Add(concurrentAttempts)
+
+	successGauge := atomic.Int32{}
+
+	for i := 0; i < concurrentAttempts; i++ {
+		go func() {
+			defer wg.Done()
+			initSuccess := cache.Init(nodeID)
+			if initSuccess {
+				successGauge.Inc()
+			}
+		}()
+	}
+
+	unittest.RequireReturnsBefore(t, wg.Wait, 100*time.Millisecond, "timed out waiting for goroutines to finish")
+
+	// ensure that only one goroutine successfully initialized the record
+	require.Equal(t, int32(1), successGauge.Load())
+
+	// ensure that the record is correctly initialized in the cache
+	guage, found, _ := cache.Get(nodeID)
+	require.True(t, found)
+	require.Zero(t, guage)
+}
+
+// TestRecordCache_Update tests the Update method of the RecordCache.
+// The test covers the following scenarios:
+// 1. Updating a record gauge for an existing node ID.
+// 2. Attempting to update a record gauge  for a non-existing node ID should not result in error. Update should always attempt to initialize the gauge.
+// 3. Multiple updates on the same record only initialize the record once.
+func TestRecordCache_Update(t *testing.T) {
+	cache := cacheFixture(t, 100, defaultDecay, zerolog.Nop(), metrics.NewNoopCollector())
+	require.NotNil(t, cache)
+	// expect cache to be initialized with a empty active cluster IDs list
+	require.Equalf(t, uint(0), cache.Size(), "cache size must be 1")
+
+	nodeID1 := unittest.IdentifierFixture()
+	nodeID2 := unittest.IdentifierFixture()
+
+	// initialize spam records for nodeID1 and nodeID2
+	require.True(t, cache.Init(nodeID1))
+	require.True(t, cache.Init(nodeID2))
+
+	guage, err := cache.Update(nodeID1)
+	require.NoError(t, err)
+	require.Equal(t, float64(1), guage)
+
+	// get will apply a slightl decay resulting
+	// in a gauge value less than guage which is 1 but greater than 0.9
+	currentGauge, ok, err := cache.Get(nodeID1)
+	require.NoError(t, err)
+	require.True(t, ok)
+	require.LessOrEqual(t, currentGauge, guage)
+	require.Greater(t, currentGauge, 0.9999999)
+
+	// test adjusting the spam record for a non-existing node ID
+	nodeID3 := unittest.IdentifierFixture()
+	guage2, err := cache.Update(nodeID3)
```

Comment:
```suggestion
	guage3, err := cache.Update(nodeID3)
```
Not required - but personally I find it easier to follow when the number of different variables is consistent: "gauge N is the gauge for node ID N"

Reply:
https://github.com/onflow/flow-go/pull/4365/commits/c523358691b76a89c49541cc90fd74a76437b867